src/TLS.cpp

```diff
@@ -182,6 +182,9 @@ ::Code SocketTLS::interpretLibCode(const int code, const bool processed, const b
 		case SSL_ERROR_WANT_WRITE:
 			return WaitOut;
 		case SSL_ERROR_SYSCALL:
+			if (ERR_get_error() == 0 && errno == EAGAIN) {
```

Comment:
I would use `osError()` instead of `errno`: https://github.com/mumble-voip/libmumble/blob/ea53abd241fa05633dcf788acd566aa576427de3/src/Socket.cpp#L211-L217

As for handling `EAGAIN`, there appears to be a mistake in `osErrorToCode()`: https://github.com/mumble-voip/libmumble/blob/ea53abd241fa05633dcf788acd566aa576427de3/src/Socket.hpp#L116-L123

`Code::Retry` should probably be returned instead.

Reply:
Seems like an abstraction leaked here :)  
But I may be wrong, and not understand code code here  

After suggested changes we start to mixing of mumble::SocketTLS::Code and mumble::codes

I think you understand better than me and can fix this place to fit your conventions

Reply:
No worries, it would be just to use the stuff we already have in the class.

Once the potential bug in `osErrorToCode()` is fixed, the new line can be:

```suggestion
			if (ERR_get_error() == 0 && osErrorToCode(osError()) == Socket::Code::Retry) {
```

Maybe we should eventually merge the `Code` enum in `SocketTLS` with the global one...